packages/client/src/execution/vmexecution.ts

```diff
@@ -676,10 +687,11 @@ export class VMExecution extends Execution {
       })
 
       if (txHashes.length === 0) {
+        this._statsVm = vm
+
         // we are skipping header validation because the block has been picked from the
         // blockchain and header should have already been validated while putBlock
         const beforeTS = Date.now()
-        this.stats(vm)
```

Comment:
One thing to note about this change is that now stat collection and logging isn't being included in the time measurement of `runBlock`, so there will potentially be a difference in the `time` value being logged for block execution.